nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/HostedHandlers.java

```diff
@@ -69,4 +69,9 @@
     context.getRepository().facet(RHostedFacet.class).upload(path, context.getRequest().getPayload());
     return HttpResponses.ok();
   };
+
+  /**
+   * Handle upload non-R file request.
+   */
+  final Handler nonRFileUpload = context -> HttpResponses.badRequest("Non-R file extension or wrong upload path.");
```

Comment:
maybe nonRArchive

Reply:
Done in 47bd5fb9dc9524a372cb0a1cd8218dad8aa45f03